plugins/ltac2/tac2intern.ml

```diff
@@ -1670,6 +1712,35 @@ let debug_globalize_allow_ext ids tac =
   let tacext ?loc (RawExt (tag,arg)) = CAst.make ?loc @@ CTacExt (tag,arg) in
   globalize_gen ~tacext ids tac
 
+let { Goptions.get = typed_notations } =
+  Goptions.declare_bool_option_and_ref
+    ~key:["Ltac2";"Typed";"Notations"] ~value:true ()
```

Comment:
I expect that typed notations will be useful most of the time, so I think it would be better to allow disabling the type-checking locally, via an attribute. Have you considered this option?

Reply:
That would mean keeping the untyped data around too, it's not great.
Also typed notations can do more things, like have `constr:()` in them.
Also I'm pretty sure untyped notations are useless.

Reply:
Ah OK, I thought this would be as simple as  using the `UntypedNota` constructor if the attribute is used, and `TypedNota` otherwise (determined at the definition time of the notation).

Untyped notations are probably mostly useless, but there might be hacky cases where they could be useful I guess.

Reply:
I guess I misunderstood what you were saying, I thought you meant something like 
~~~coq
Ltac2 Notation "foo" ...

#[thing]
Ltac2 whatevr := foo ...
~~~
and in the definition `foo` would be treated as an untyped notation, but maybe you mean just having 
~~~coq
#[thing] Ltac2 Notation "foo" ...
~~~
as an alternative to
~~~coq
Unset Ltac2 Typed Notations.
Ltac2 Notation "foo" ...
~~~

Reply:
I indeed meant the latter, that is, allowing something like:
```
#[untyped]
Ltac2 Notation ...
```